packages/frontend-2/middleware/requireValidProject.ts

```diff
@@ -28,14 +28,18 @@ export default defineNuxtRouteMiddleware(async (to) => {
   const { isLoggedIn } = useActiveUser()
   const isWorkspacesEnabled = useIsWorkspacesEnabled()
 
+  // if same path and query, lets skip refetch - its likely a viewer hash update
+  const isSamePathAndQuery =
+    to.path + JSON.stringify(to.query) === from.path + JSON.stringify(from.query)
+
   const { data, errors } = await client
     .query({
       query: projectAccessCheckQuery,
       variables: { id: projectId },
       context: {
         skipLoggingErrors: true
       },
-      fetchPolicy: 'network-only'
+      fetchPolicy: isSamePathAndQuery ? 'cache-first' : 'network-only'
```

Comment:
cc @Mikehrn @andrewwallacespeckle 

turns out every change to the querystring _or_ the anchor/hash state (e.g. #threadId in the viewer page) causes all of the middlewares to refire, cause technically its a new navigation.

this causes a lot of the queries to re-fire, especially if they have the `network-only` fetchPolicy (which would otherwise make sense). so we have to explicitly check in middlewares if the navigation isn't actually just a minor querystring/anchor string change that shouldn't actually cause full access rechecking w/ refreshed queries